Fix get Current/get Currents using stateExt and iCurrentControl

lrapetti opened the pull request:
Fix the implementation of `getCurrent` and `getCurrents` methods in the `RemoteControlBoard` using the `stateExt:o` port.

The implementation has been modified also in the `ControlBoardWrapper` in order to use the `ICurrentControl` interface methods.

barbalberto commented:
Looks good to me. @randaz81 the `getCurrent` from `iAmplifier` and from `iCurrentControl` interface are equivalent, right?

Nicogene commented:
Please add a line in the changelog `v3_0_1.md` thanks!

lrapetti commented:
@Nicogene 
changelog updated https://github.com/robotology/yarp/pull/1769/commits/4891fc20aaa9f41ef9569551579ef30acdb98a3a

lrapetti commented:
>  could you please rebase the branch over the latest devel

@Nicogene 
Did you mean the latest master?

Nicogene commented:
>Did you mean the latest master?
Yes sorry it was a lapsus!

barbalberto commented:
I'd suggest a small change. The two functions `getCurrent` in the `amp` interface and in the `iCurrentControl` are equivalent. The low level device may or may not implement any of them.

Since we are improving the wrapper, we may use the `iCurrentControl` as you did and, if the interface is not implemented by the device, the wrapper can fallback to the `amp` interface and try to get the values from there.

Anyway great catch!! Well done

lrapetti commented:
> Since we are improving the wrapper, we may use the iCurrentControl as you did and, if the interface is not implemented by the device, the wrapper can fallback to the amp interface and tries to get the values from there.

@barbalberto 
I have add the use of `IAmplifier` in case the `ICurrentControl` is not found, so both interfaces are now supported.

barbalberto commented:
@lrapetti I'd like to ask you another small change.

Suppose the device has both `amp` and `current` interfaces implemented, and the `getCurrent` method returns false.

In this case the `getCurrent` method will be called twice, and the second time is useless because we already know the result will be false, because they collapse to the same code in the low level device.

The device is, supposely, getting the values from the hardware and it may takes some time to produce the answer.

I'd split the check in two parts
```
if(interface1)
    getCurrent
else if(interface2)
    getCurrent

if(ret)
   assign values    // just one for loop

```
This way the `getCurrent` is called only once.

lrapetti commented:
@barbalberto I have updated the commit as you suggested and rebased.
According to the changes I have changed the error condition initializing `ret=false`:
```
ret = false
if(interface1)
    ret = getCurrent
else if(interface2)
    ret = getCurrent

if(ret)
   assign values 
else
   error condition
```

barbalberto commented:
Great, many thanks. When test are succesfully we can merge it
